File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        seen = {}

        for i in range(len(nums)):
            
            num1 = nums[i]
            complement = target - num1
            if complement in seen:
                return [seen[complement], i]

            seen[num1] = i

        return False
        

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int len = nums.length;
        if (len == 0)
            return new int[] {};
        int[] lt = new int[len];
        int[] rt = new int[len];
        int[] sol = new int[len];
        lt[0] = 1;
        rt[len - 1] = 1;

        for (int i = 1; i < len; i++) {
            lt[i] = nums[i - 1] * lt[i - 1];
        }
        for (int i = len-2; i >= 0; i--) {
            rt[i] = nums[i +1 ] * rt[i + 1];
        }
        for (int i = 0; i < len; i++) {
            sol[i] = lt[i] * rt[i];
        }
        return sol;
    }
}
